dispatch_json/bundle/main.ts

```diff
@@ -84,3 +83,34 @@ export async function sendAsync(
   const res = await promise;
   return unwrapResponse(res);
 }
+
+export class JsonOp {
+
+  private opId?: number; 
+  
+  constructor(
+    private readonly opNamespace: string,
+    private readonly opName: string,
+  ) {
+    ops[this.opNamespace][this.opName] = (id?: number): void => {
+      this.opId = id;
+      if (id !== undefined) {
```

Comment:
This should throw if `id === undefined` (not sure if that case is possible).

Reply:
`undefined` is used to signal a "reset" to listeners, since I can't really `unsetAsyncHandler` currently we just ignore. The only time a "reset" happens is when you call `Isolate::set_dispatcher_registry`(only when creating a new worker in our case). `setAsyncHandler` has no way to "correctly" handle a `opId` value that isn't a number, thus passing `undefined` will throw a error.